src/drive/mobile/containers/UploadProgression.jsx

```diff
@@ -1,13 +1,87 @@
+import React from 'react'
 import { connect } from 'react-redux'
-import UploadProgression from '../components/UploadProgression'
-
-const mapStateToProps = state => ({
-  message: state.mobile.mediaBackup.currentUpload
-    ? state.mobile.mediaBackup.currentUpload.message
-    : undefined,
-  messageData: state.mobile.mediaBackup.currentUpload
-    ? state.mobile.mediaBackup.currentUpload.messageData
-    : undefined
-})
-
-export default connect(mapStateToProps)(UploadProgression)
+import classnames from 'classnames'
+import styles from '../styles/uploadprogression'
+import { translate } from 'cozy-ui/react/I18n'
+
+const percent = (actual, total) => Math.floor(actual * 100 / total)
+
+const Progress = ({ percent, style, color, background }) => {
+  const containerStyle = {
+    opacity: percent < 100 ? 1 : 0,
+    transition: `${0.4}s opacity}`,
+    transitionDelay: `${percent < 100 ? 0 : 0.4}s`,
+    height: `${2}px`,
+    backgroundColor: background ? 'lightgray' : ''
+  }
+  const barStyle = {
+    display: 'inline-block',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: `${percent}%`,
+    maxWidth: '100%',
+    height: `${2}px`,
+    borderRadius: '0 1px 1px 0',
+    transition: `${0.4}s width, ${0.4}s background-color`,
+    backgroundColor: color || 'darkblue',
+    ...style
+  }
+  return (
+    <div style={containerStyle}>
+      <div style={barStyle} />
+    </div>
+  )
+}
+
+const UploadProgression = ({ t, current, total, media }) => {
+  return (
+    <div className={styles['coz-upload-status']}>
+      <Progress percent={percent(current, total)} />
+      {media.filePath ? (
+        <img className={styles['coz-progress-pic']} src={media.filePath} />
+      ) : (
+        <div
+          className={styles['coz-progress-pic']}
+          style={{ border: '1px solid darkgray', backgroundColor: 'lightGray' }}
```

Comment:
Why are we using a combo of classes and inline styles here?

Reply:
But… `coz-progress-pic` is also not a global css rule, is it? It's part of the local css module.

On principle, I absolutely agree that this should be how we do things (albeit with utility classes and not inline styles). But it's not what we've been doing so far, am I don't think it's a good idea to start changing it here, without any global effort or even discussion.

> Let's say we want to change this square with a placekitten picture or any other default image with a real binary file. Do we want to change a css file? Or do we want to have everything in the same place?

The thing is, we're going to start looking in the css files, because that's where all of our styles are. I think the correct approach (because it would be consistent) here is to introduce a modifier class (`.coz-progress-pic.--placeholder` or something similar) in the css.